novice-goals.Rmd

```diff
@@ -0,0 +1,226 @@
+# Outline {#outline}
+
+This outline describes the questions that the novice courses on R and Python will answer.
+The advanced course can then assume that learners have hands-on experience with these topics
+but nothing more.
+
+## Getting started
+
+-   What are the different ways I can interact with software?
+    -   console
+    -   scripts
+-   How can I find and view help?
+    -   In the IDE
+    -   Stack Overflow
+-   How can I inspect data while I'm working on it?
+    -   table viewers
+-   Where does stuff actually live on my computer?
+    -   directory structure on Windows and Unix
+    -   absolute vs. relative paths
+    -   data on disk vs. data in memory
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   what CSV is, where it comes from, and why people use it
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   arrange
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+    -   the difference between syntax and runtime errors
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+    -   split-apply-combine
+-   How can I work with two or more datasets?
+    -   join
+-   How can I save my results?
+    -   writing files
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
+    -   loops and conditionals
+
+## Plotting
+
+-   Why plot?
+    -   summary statistics can mislead
+    -   [Anscombe's Quartet and the DataSaurus dozen][anscome-datasaurus]
+-   What are the core elements of every plot?
+    -   data
+    -   geometric objects
+    -   aesthetic mapping
+-   How can I create different kinds of plots?
+    -   scatter plot
+    -   line plot
+    -   histogram
+    -   bar plot
+    -   which to use when
+-   How can I plot multiple datasets at once?
+    -   grouping
+    -   faceting
+-   How can I make misleading plots?
+    -   showing a single central tendency data point instead of the individual observations
+    -   saturated plots instead of for example violins or 2D histograms
+    -   not starting axis at zero to exaggerate a difference
+    -   not using perceptually uniform colormaps to indicate quantities
```

Comment:
This comment and the one below would be good to discuss e.g. viridis color scale.

Reply:
Yes, [the matplotlib docs](https://matplotlib.org/users/colormaps.html) has great info that can be simplified and abbreviated.